gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -291,9 +366,49 @@ public int read(ByteBuffer byteBuffer) throws IOException {
         bufferedContentReadOffset = bytesToWrite;
       }
     }
+
+    if (footerContent != null && position >= footerStartOffsetInBytes && byteBuffer
```

Comment:
I'd be happy to see what the purpose of this condition is by using a local variable (`hasMoreFooterToRead`-like?) that would make this long-ish `if` easier.

Reply:
Agreed